### ISSUE TITLE
Debugger Functionality

### ISSUE THREAD

**Issue by Manuzor** [f1nalspace/final_game_tech#3]
As a windows developer that uses the debugger a lot, it would be nice to have the following available in a portable way.

```
__debugbreak(); // Should be a macro so the call stack is on the correct line when executed.
BOOL IsDebuggerPresent();
void OutputDebugString(const TCHAR* zStr);
```

This applies to the `final_platform_layer.h` part of this repo, of course.

**Comment by f1nalspace** [comment]
Debugbreak is already implemented, Debuggerpresent will come for Win32 and Linux. But for OutputDebugString there is no equivalent on UNIX/Linux :-( We could use the console output for non-win32, but that feels not right.

**Comment by Manuzor** [comment]
Thanks for your effort!

Using console output for non-win32 is probably not a good idea. Take this piece of code in the context of a console application for example
```c
void fatal_error(const char* message)
{
    fplConsoleError(message); // print to stderr
    fplDebugOut(message); // print to debugger stream
}
```

On non-windows platforms, this would print all fatal error message twice in a row. 😟 

Maybe `fplDebugOut` just shouldn't do anything on non-windows platforms? 🤔

I also thought about always routing `fplConsoleOut/Error` message through `OutputDebugString` as well as `stdout/stderr`, but if the user wants this kind of behavior, it's easy enough for them to just define something like `fatal_error` above that does the job. I think `fpl` should not dictate this kind of behavior. I could imagine that some users might want to be in charge of where the different kinds of messages end up in, e.g. dev-only infos go through `OutputDebugString`, and user-facing output through `stdout/stderr`.

**Comment by f1nalspace** [comment]
I will add fpLDebugOut and use OutputDebugString for windows all others does nothing.
Also i will change the behavior about error reporting a bit, so it can be more controlled.

**Comment by f1nalspace** [comment]
In the develop branch i refactored the logging system a lot, so in the next release you can configure logging however you want. I will close down this issue when its released.

**Comment by f1nalspace** [comment]
Issue is solved as of release 0.8.1.
